src/NLPModels.jl

```diff
@@ -282,6 +286,7 @@ end
 "Evaluate the product of the Lagrangian Hessian at `(x,y)` with the vector `v` in place."
 function hprod!(nlp :: NLPModel, x :: Array{Float64}, y :: Array{Float64}, v :: Array{Float64}, hv :: Array{Float64})
   nlp.neval_hprod += 1
+  hv = zeros(size(x)) # JPD workaround 
```

Comment:
This creates a new vector of zeros. Wouldn't it be enough to fill the `hv` supplied by the user with zeros?